tests/test_check.py

```diff
@@ -18,31 +18,31 @@
 from twine.commands import check
 
 
-def test_warningstream_write_match():
-    stream = check._WarningStream()
-    stream.output = pretend.stub(write=pretend.call_recorder(lambda a: None))
+class TestWarningStream:
```

Comment:
I *think* this is the first instance of using a test class in Twine. Maybe it's worth doing more that, though here it seems like the primary benefit is for the `setup` of `self.stream`. Did you consider using a fixture?

```python
@pytest.fixture
def warning_stream():
    stream = check._WarningStream()
    stream.output = pretend.stub(
        write=pretend.call_recorder(lambda a: None),
        getvalue=lambda: "result",
    )
    return stream


def test_warningstream_write_nomatch(warning_stream):
    # ...
```

Reply:
> Did you consider using a fixture?

Not really.

Since these are all tests of a single class's behavior, initializing the object to be tested in a fixture doesn't feel intuitive to me.

I think my mental model is to use fixtures when the setup is going to be used for things like "need temporary directory" or "setup isolated environment" -- tests wouldn't check the behavior of the fixture-object.

Reply:
All that said, if folks feel strongly about avoiding use of test classes, I'm happy to change this up.

Reply:
This is surely good enough, for sure. If someone feels strongly, we can refactor later.